src/util/Xhr.js

```diff
@@ -441,10 +525,11 @@ class Xhr {
                     this.xhr.upload.addEventListener('progress', progressHandler);
                 }
 
-                if (formData) {
-                    this.xhr.send(formData);
+                const dataSent = formData || data;
+                if (withIdleTimeout) {
+                    xhrSendWithIdleTimeout(this.xhr, dataSent, idleTimeoutDuration, idleTimeoutHandler);
```

Comment:
What does the webapp do here when 120s has passed with no progress?

Reply:
it'll retry uploading the chunk unless the retry limit has been exceeded